### ISSUE TITLE
Bug Report: Selecting really quickly --> Crash

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#2591]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable): 0.4

Any other software?
```

# Steps to reproduce

1. make a selection (left-click + drag)
2. right-click to copy
3. repeat 1 and 2 a lot and very quickly

I specifically tried to have the selections be big. Not sure if that matters.

# Expected behavior

copy then erase the selection over and over again

# Actual behavior

```
 	TerminalControl.dll!wil::details::WilDynamicLoadRaiseFailFastException(_EXCEPTION_RECORD * er, _CONTEXT * cr, unsigned long flags) Line 1731	C++
 	TerminalControl.dll!wil::details::WilRaiseFailFastException(_EXCEPTION_RECORD * er, _CONTEXT * cr, unsigned long flags) Line 1671	C++
 	TerminalControl.dll!wil::details::WilFailFast(const wil::FailureInfo & failure) Line 3352	C++
 	TerminalControl.dll!wil::details::ReportFailure(void * callerReturnAddress, unsigned int lineNumber, const char * fileName, const char * functionName, const char * code, void * returnAddress, wil::FailureType type, HRESULT hr, const wchar_t * message, wil::details::ReportFailureOptions options) Line 3373	C++
 	TerminalControl.dll!wil::details::ReportFailure_Hr(void * callerReturnAddress, unsigned int lineNumber, const char * fileName, const char * functionName, const char * code, void * returnAddress, wil::FailureType type, HRESULT hr) Line 3451	C++
 	TerminalControl.dll!wil::details::in1diag5::_FailFast_Hr(void * callerReturnAddress, unsigned int lineNumber, const char * fileName, const char * functionName, const char * code, HRESULT hr) Line 4287	C++
 	TerminalControl.dll!wil::details::in1diag5::FailFast_IfFailed(void * callerReturnAddress, unsigned int lineNumber, const char * fileName, const char * functionName, const char * code, HRESULT hr) Line 4319	C++
 	TerminalControl.dll!Microsoft::Console::Render::DxEngine::Present() Line 836	C++
 	TerminalControl.dll!Microsoft::Console::Render::Renderer::_PaintFrameForEngine(Microsoft::Console::Render::IRenderEngine * const pEngine) Line 130	C++
 	TerminalControl.dll!Microsoft::Console::Render::Renderer::PaintFrame() Line 65	C++
 	TerminalControl.dll!Microsoft::Console::Render::RenderThread::_ThreadProc() Line 165	C++
>	TerminalControl.dll!Microsoft::Console::Render::RenderThread::s_ThreadProc(void * lpParameter) Line 148	C++
```

Also this happens:
![image](https://user-images.githubusercontent.com/11050425/63896197-3310bd80-c9a6-11e9-834e-00033ad23ac2.png)

**Comment by tobiaskohlbau** [comment]
I ran into this issue quit frequently. During a work day I hit this issue 5-10 times, whereas the terminal closes the selection isn't copied into clipboard. Let me know if there is anything I can help with to debug this issue.

**Comment by DHowett-MSFT** [comment]
This call stack points to an issue very similar to the one fixed by #2830. I'll try to repro it now that 2830 has merged.

**Comment by tobiaskohlbau** [comment]
I'm unable to reproduce this with the latest store released version. LGTM
